File: Heat/ViewModels/Plants/DetailsThermalPlantViewModel.vb
Imports System.ComponentModel.DataAnnotations


Namespace ViewModels.Plants
    ''' <summary>
    ''' Oggetto con ottenuto con flattening parziale di Plant, Plant.BuildingAddress e Plant.ThermalUnit
    ''' </summary>
    ''' <remarks></remarks>
    Public Class DetailsThermalPlantViewModel

#Region "Proprietà derivate da Plant"
        Property ID As Integer
        <Display(name:="Classe impianto")> _
        Property PlantClass As String

        <Display(name:="Tipologia impianto")> _
        Property PlantType As String

#End Region

#Region "Proprietà derivate da Plant.BuidingAddress"
        <Display(name:="Singola unità abitativa")> _
        Property IsSingleUnit As Boolean

        <Display(name:="Categoria energetica")> _
        Property EnergyCategory As EnergyCategoryEnum

        <Display(name:="Volume lordo riscaldato (m³)")> _
        Property GrossHeatedVolumeM3 As Single

        <Display(name:="Volume lordo raffrescato (m³)")> _
        Property GrossCooledVolumeM3 As Single
#End Region

#Region "Proprietà derivate da Plant.ThermalUnit"
        <Display(name:="Costruttore")>
        Property Manifacturer As String

        <Display(name:="Modello")> _
        Property Model As String

        <Display(name:="Matricola")> _
        Property SerialNumber As String

        <Display(name:="Data di installazione")> _
        <DataType(DataType.Date)> _
        <DisplayFormat(dataformatstring:="{0:d}")> _
        Property InstallationDate As DateTime

        <Display(name:="Data di prima accensione")> _
        <DataType(DataType.Date)> _
        <DisplayFormat(dataformatstring:="{0:d}")> _
        Property FirstStartUp As Nullable(Of DateTime)

        <Display(name:="Garanzia")> _
        Property Warranty As String

        <Display(name:="Scadenza garanzia")> _
        <DataType(DataType.Date)> _
        <DisplayFormat(dataformatstring:="{0:d}")> _
        Property WarrantyExpiration As Nullable(Of DateTime)

        <Display(name:="Combustibile")> _
        Property Fuel As String

        <Display(name:="Potenza termica utile nominale Pn max (kW)")> _
        Property NominalThermalPowerMax As Single

        <Display(name:="Fluido termovettore")> _
        Property HeatTransferFluid As String

        <Display(name:="Rendimento termico utile a Pn max (%)")> _
        Property ThermalEfficiencyPowerMax As Single

        <Display(name:="Tipo di unità termica")> _
        Property Kind As ThermalUnitKindEnum

        'Property Heaters As List(Of Heater)
#End Region

    End Class
End Namespace

File: Heat.ConvertedToC#/ViewModels/Plants/DetailsThermalPlantViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;


namespace Heat.ViewModels.Plants
{
    /// <summary>
    /// Oggetto con ottenuto con flattening parziale di Plant, Plant.BuildingAddress e Plant.ThermalUnit
    /// </summary>
    /// <remarks></remarks>
    public class DetailsThermalPlantViewModel
	{

		#region "Proprietà derivate da Plant"
		public int ID { get; set; }
		[Display(Name = "Classe impianto")]
		public string PlantClass { get; set; }

		[Display(Name = "Tipologia impianto")]
		public string PlantType { get; set; }


		#endregion

		#region "Proprietà derivate da Plant.BuidingAddress"
		[Display(Name = "Singola unità abitativa")]
		public bool IsSingleUnit { get; set; }

		[Display(Name = "Categoria energetica")]
		public EnergyCategoryEnum EnergyCategory { get; set; }

		[Display(Name = "Volume lordo riscaldato (m³)")]
		public float GrossHeatedVolumeM3 { get; set; }

		[Display(Name = "Volume lordo raffrescato (m³)")]
		public float GrossCooledVolumeM3 { get; set; }
		#endregion

		#region "Proprietà derivate da Plant.ThermalUnit"
		[Display(Name = "Costruttore")]
		public string Manifacturer { get; set; }

		[Display(Name = "Modello")]
		public string Model { get; set; }

		[Display(Name = "Matricola")]
		public string SerialNumber { get; set; }

		[Display(Name = "Data di installazione")]
		[DataType(DataType.Date)]
		[DisplayFormat(DataFormatString = "{0:d}")]
		public DateTime InstallationDate { get; set; }

		[Display(Name = "Data di prima accensione")]
		[DataType(DataType.Date)]
		[DisplayFormat(DataFormatString = "{0:d}")]
		public Nullable<DateTime> FirstStartUp { get; set; }

		[Display(Name = "Garanzia")]
		public string Warranty { get; set; }

		[Display(Name = "Scadenza garanzia")]
		[DataType(DataType.Date)]
		[DisplayFormat(DataFormatString = "{0:d}")]
		public Nullable<DateTime> WarrantyExpiration { get; set; }

		[Display(Name = "Combustibile")]
		public string Fuel { get; set; }

		[Display(Name = "Potenza termica utile nominale Pn max (kW)")]
		public float NominalThermalPowerMax { get; set; }

		[Display(Name = "Fluido termovettore")]
		public string HeatTransferFluid { get; set; }

		[Display(Name = "Rendimento termico utile a Pn max (%)")]
		public float ThermalEfficiencyPowerMax { get; set; }

		[Display(Name = "Tipo di unità termica")]
		public ThermalUnitKindEnum Kind { get; set; }

		//Property Heaters As List(Of Heater)
		#endregion

	}
}

